src/coap_io.c

```diff
@@ -401,6 +401,7 @@ coap_socket_bind_tcp(coap_socket_t *sock,
 #endif
     coap_log(LOG_WARNING, "coap_socket_bind_tcp: ioctl FIONBIO: %s\n", coap_socket_strerror());
   }
+  setsockopt (sock->fd, SOL_SOCKET, SO_KEEPALIVE, &on ,sizeof (on));
```

Comment:
Why is the `OPTVAL_T` around `&on` not needed here?
Are keepalives entirely optional so that the success of the setsockopt call is irrelevant?

Reply:
Code corrected to model other setsockopt() calls.